src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -225,8 +220,26 @@ public function has($id)
      */
     public function get($id, $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE)
     {
+        $serviceNotFound = function ($id) use ($invalidBehavior) {
```

Comment:
I don't think this is correct. Perf wise, this will create a closure for each "get". But get is a perf critical piece, so that it should do as little as possible on each code path.